xcm/xcm-emulator/src/lib.rs

```diff
@@ -1107,8 +1240,127 @@ macro_rules! decl_test_sender_receiver_accounts_parameter_types {
 	};
 }
 
+pub struct DefaultMessageProcessor<T>(PhantomData<T>);
+impl<T> ProcessMessage for DefaultMessageProcessor<T>
+where
+	T: Chain + RelayChain,
+	T::Runtime: MessageQueueConfig,
+	<<T::Runtime as MessageQueueConfig>::MessageProcessor as ProcessMessage>::Origin:
+		PartialEq<AggregateMessageOrigin>,
+	MessageQueuePallet<T::Runtime>: EnqueueMessage<AggregateMessageOrigin> + ServiceQueues,
+{
+	type Origin = ParaId;
+
+	fn process_message(
+		msg: &[u8],
+		para: Self::Origin,
+		_meter: &mut WeightMeter,
+		_id: &mut XcmHash,
+	) -> Result<bool, ProcessMessageError> {
+		MessageQueuePallet::<T::Runtime>::enqueue_message(
+			msg.try_into().expect("Message too long"),
+			AggregateMessageOrigin::Ump(UmpQueueId::Para(para)),
+		);
+		MessageQueuePallet::<T::Runtime>::service_queues(Weight::MAX);
+
+		Ok(true)
+	}
+}
+
+#[derive(Clone)]
+pub struct TestAccount {
+	pub account_id: AccountId,
+	pub balance: Balance,
+}
+
+#[derive(Clone)]
+pub struct TestArgs {
```

Comment:
without docs this crate does not get more friendly for users, even with refactorings

Reply:
Ok, I'll add some. I still want to create good docs in a README, I kind of integration tests developer guide that will help with it.